### PR TITLE
Fix some compiler warnings

### PR DESCRIPTION
**Pull request by nsivov**
Compiled with gcc 8.2.0/x86_64-linux-gnu.

### CODE REVIEW

File: mono/utils/mono-state.c [hunk 1]
```diff
@@ -116,8 +116,13 @@ mono_summarize_toggle_assertions (gboolean enable)
 	}
 
 	mono_memory_barrier ();
-#endif
 }
```
Comment (baulig) [inline mono/utils/mono-state.c]: What's the reason behind this change?  I think it looks more ugly when we have the same function defined both inside and outside the conditional.

File: mono/utils/mono-state.c [hunk 2]
```diff
@@ -687,6 +692,7 @@ mono_native_state_add_thread (MonoStateWriter *writer, MonoThreadSummary *thread
 static void
 mono_native_state_add_ee_info  (MonoStateWriter *writer)
 {
+#ifndef MONO_PRIVATE_CRASHES
```
Comment (baulig) [inline mono/utils/mono-state.c]: Where does this conditional come from?  This particular change also seems to modify behavior, so it's more than just fixing compiler warnings.

Comment (nsivov) [reply mono/utils/mono-state.c]: As far as I can tell it comes from configure, controlled with --with-crash_privacy=yes,no. What behavior does this change modify? It turns existing function to empty body, and changes to call it unconditionally, to fix unused function warning.

File: mono/utils/mono-state.c (hunk 1 above)
Comment (nsivov) [reply mono/utils/mono-state.c]: The reason is to move unused *_not_reached_* functions under conditional block. I was thinking one block is better than two for the same condition, but surely I can keep function intact if you prefer.

File: mono/utils/mono-state.c (hunk 2 above)
Comment (lewurm) [reply mono/utils/mono-state.c]: this is fine

File: mono/utils/mono-state.c (hunk 1 above)
Comment (lewurm) [reply mono/utils/mono-state.c]: I agree with @baulig here, we use that style everywhere and even you do it in `mono_native_state_add_ee_info` below 🙂

**Comment by lewurm** [comment]
@monojenkins build Linux AArch64 Coop Suspend

**Comment by lewurm** [comment]
@monojenkins squash

File: mono/utils/mono-state.c (hunk 1 above)
Comment (nsivov) [reply mono/utils/mono-state.c]: Sure, I'll keep that in mind. To me the difference was about having conditional expression in one place or in two, having it consolidated seemed better.
